### PR TITLE
GFORMS-2714 - Add ability to reference columns in lookup/csv - For lo…

### PR DESCRIPTION
**Pull request by DursunGundogan**
…okup(port) and lookup(vfrsTradeSector) plus make easier to implement for other lookups

### CODE REVIEW

File: app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala [hunk 1]
```diff
@@ -1093,47 +1093,6 @@ object FormTemplateValidator {
...
-              for {
-                dataRetrieve <- page.dataRetrieves()
-                iff          <- dataRetrieve.`if`.toList
-                result       <- verifyIncludeIf(idx).apply(iff.booleanExpr)
-              } yield result
-
-            verifiedValidIf ++ verifiedIncludeIf ++ verifiedComponentIncludeIfs ++ verifiedDataRetrieveIncludeIf ++ verifiedRemoveItemIf
-          }
-      )
-  }
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala:1135]: Why is this removed?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala:1135]: This change ensures forward references are allowed in if , validIf and includeif expressions.

**Review by josef-vlach — APPROVED** [review]
